### ISSUE TITLE
Run exe file after downloading

### ISSUE THREAD

**Issue by green13** [webtorrent/webtorrent#198]
Hello. 
I use webtorrent in node, os windows 8.1.
I'm trying to run downloaded .exe file (webtorrent is running) and get the error "file is locked by another process".

**Comment by green13** [comment]
Any news on this?

**Comment by arestov** [comment]
I guess that the reason is you need `write file descriptor (fd)` which block file reading, but after loading finish you need only `read` `fd` rights . So after finish we must free `write` `fd` (maby even before `finish` event).

Real kung fu is to request when we will request `write rd` only when we really need to write and close if we dont need, by using throtling by data length and time (for e.g. not often than 500kb and not ofter than 0.5s)

**Comment by green13** [comment]
Call `torrent.store.close` helps.

**Comment by ThaUnknown** [comment]
this is a problematic issue to fix, TLDR for the chunk store to be... fast and not loose data during crashes, outages etc in performs in place writes and reads, those however require a [file lock](https://en.wikipedia.org/wiki/File_locking), the only way to not do this, is to do what the [FSA](https://web.dev/file-system-access/) does and do file swapping, aka you write data to a file with a different name, then once you close the file it swaps the filename for the wanted one, aka .crswap but this has an insane amount of issues, you use x3 as much storage while the file write is open, cannot perform in place writes and reads and have high chances of data loss or data duplication, so I do not see any other way of solving this issue without closing the store or the torrent

TLDR: if you want to run an .exe you need to destroy the torrent or close the store, or copy the file to another location and run if from there, there's no other clean way of solving this without sacrificing performance, reliability and speed
